### PR TITLE
Feature/#66 Converter between Image and BufferedImage

### PR DESCRIPTION
**Pull request by dkrawiec**
#66

### CODE REVIEW

**Comment by dkrawiec** [comment]
@kmisztal could you please review the changes?

File: imageprocessor/imageprocessor-awt/src/main/java/pl/edu/uj/kimage/awt/ImageConverter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                bufferedImage.setRGB(i, j, getIntFromColor(argb.getAlpha(), argb.getRed(), argb.getGreen(), argb.getBlue()));  
+            }
+        }
+        
+        return bufferedImage;
+    }
+    
+    
+  private int getIntFromColor(int alpha, int red, int green, int blue){
+    red = (red << 16) ;
```
Comment (kmisztal) [inline imageprocessor/imageprocessor-awt/src/main/java/pl/edu/uj/kimage/awt/ImageConverter.java]: red = (red << 16) & 0x00FF0000;
green = (green << 8) & 0x0000FF00; 
blue = blue & 0x000000FF;
alpha =  (alpha << 24) & 0xFF000000;

this looks more safer

File: imageprocessor/imageprocessor-awt/src/test/java/pl/edu/uj/kimage/awt/ImageConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+                assertThat((argb >> 16) & 0x000000FF).isEqualTo(image.getColor(i, j).getRed());
+                assertThat((argb >> 8) & 0x000000FF).isEqualTo(image.getColor(i, j).getGreen());
+                assertThat((argb) & 0x000000FF).isEqualTo(image.getColor(i, j).getBlue());
+                assertThat((argb >> 24) & 0x000000FF).isEqualTo(image.getColor(i, j).getAlpha());
+            }
+        }
+    }
+
+}
```
Comment (kmisztal) [inline imageprocessor/imageprocessor-awt/src/test/java/pl/edu/uj/kimage/awt/ImageConverterTest.java:78]: ok

File: imageprocessor/imageprocessor-awt/src/main/java/pl/edu/uj/kimage/awt/ImageConverter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import pl.edu.uj.kimage.plugin.model.Color;
+import pl.edu.uj.kimage.plugin.model.Image;
+import java.awt.image.BufferedImage;
+
+
+public class ImageConverter {
+    
+     public Image toImage(BufferedImage bufferedImage) {
+          
+        int width = bufferedImage.getWidth();
```
Comment (kmisztal) [inline imageprocessor/imageprocessor-awt/src/main/java/pl/edu/uj/kimage/awt/ImageConverter.java]: width and height are automatically set to final?

File: imageprocessor/imageprocessor-plugin/src/main/java/pl/edu/uj/kimage/plugin/model/Image.java [hunk 4]
```diff
@@ -6,7 +6,7 @@
     private final int width;
     private final int height;
 
-    Image(final int width, final int height, final Color[] data) {
+    public Image(final int width, final int height, final Color[] data) {
```
Comment (kmisztal) [inline imageprocessor/imageprocessor-plugin/src/main/java/pl/edu/uj/kimage/plugin/model/Image.java]: ok

**Pull request merged**